intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+# An introduction to calculating Cred and Grain
+
+I’m starting from the assumption that you have some familiarity with the SourceCred project. If you don’t I’d suggest you check out Dandelion's [20 minute talk (+Qs) from SustainWeb3](https://www.youtube.com/watch?v=yVTqRLekRl4), and some of the other documents in this repo.
+
+Overall, I’m going to throw around some math, but you should be able to get solid intuition on this if you’re comfortable talking about a [graph](https://en.wikipedia.org/wiki/Graph_theory) as a set of nodes and edges connecting them, where the edges can be directed and have weights. I’ll also talk some about [the PageRank algorithm](https://en.wikipedia.org/wiki/PageRank), for computing “importance” of nodes in a graph based on the “importance” of nodes connected to it, but I’ll try to motivate the intuition behind it as well.
+
+## The high level overview
+
+SourceCred takes information about contributions and generates a weights graph, and this post will first explain what that graph is and where it comes from. From this weights graph, it runs PageRank to generate a graph of Cred. This graph of Cred is then used to distribute grain. The grain can then be kept or traded, and soon will be able to be used to influence future graph weights. This process is diagrammed in the figure below.
+ 
+![](https://docs.google.com/drawings/u/1/d/s50mxZJ1DFD_yr4OFUxUd9g/image?w=551&h=250&rev=57&ac=1&parent=1kRxnuKAogf_cE0y6u2QYcjQimE0-jrBZ-LxcXKOUITw)
+
+> Figure 1. A very high level overview of how SourceCred allocates Cred and Grain, distinguishing between actions taken by contributors and those taken by the SourceCred algorithm.
+
+The rest of this doc outlines how collaboration data is gathered and structured, and how the exact amounts of Cred and Grain are calculated, and ends with some future directions SourceCred as a project is considering.
+
+## Components of contributing in SourceCred
+
+SourceCred first generates a weights graph from the input data. In this graph, a node is a contributor or contribution, and edges represent some sort of connection between the nodes. These connections are intended to include all possible ways that a contribution or contributor on a platform can express dependencies, thanks, or flows of value. Examples would include anything from a mention on a forum post to liking or merging a PR on GitHub.
+
+#### Nodes:
+
+Contributions are represented as nodes in the graph, and are given addresses to identify them uniquely. These addresses include a prefix, which takes the form: organization making the plugin that generates the node + name of the plugin + type of contribution (e.g. topic/post/user/like).
```

Comment:
Should takes the form: be in code so it's like:

`organization making the plugin that generates the node + name of the plugin + type of contribution`

or maybe like:

`[organization making the plugin that generates the node] + [name of the plugin] + [type of contribution]`

or a third style:

> organization making the plugin that generates the node + name of the plugin + type of contribution

Reply:
I really like the second one, and changed it to that.  Thanks!

Reply:
I'm not sure if this detailed stuff on node addresses belong in an intro document... it's relevant for someone who wants to write a plugin, but otherwise "below the fold". the UI generally hides node addresses.

Reply:
I left this for now, partly out of inertia, but partly because I liked being able to say later on that you can adjust weights by address prefix.  Feel free to remove at any point.